src/App.tsx

```diff
@@ -1,26 +1,195 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  getTodos,
+  USER_ID,
+  deleteTodo,
+  addTodo,
+  updateTodo,
+} from './api/todos';
 
-const USER_ID = 0;
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
+import { Filters } from './types/Filtres';
+import { ErrorNotification } from './components/ErrorNotification';
+import { filterTodos } from './utils/filterTodos';
+import { ErrorType } from './types/ErrorType';
+import { TodoList } from './components/TodoList';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorType>(ErrorType.Empty);
+  const [currentFilter, setCurrentFilter] = useState<Filters>(Filters.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [, setIsToggleAllActive] = useState<boolean>(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const disabledTodo = todos.every(todo => !todo.completed);
+
+  const activeTodos = todos.filter(todo => !todo.completed).length;
+  const filteredTodos = filterTodos(todos, currentFilter);
+  const areAllTodosCompleted = todos.every(todo => todo.completed);
+
+  const handleAddTodo = async (todoTitle: string) => {
+    setTempTodo({
+      id: 0,
+      title: todoTitle,
+      completed: false,
+      userId: USER_ID,
+    });
+    try {
+      const newTodo = await addTodo({ title: todoTitle, completed: false });
+
+      setTodos(prev => [...prev, newTodo]);
+    } catch (err) {
+      setErrorMessage(ErrorType.AddTodo);
+      throw err;
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const handleDeleteTodo = async (todoId: number) => {
+    setLoadingTodoIds(prev => [...prev, todoId]);
+    try {
+      await deleteTodo(todoId);
+
+      setTodos(prev => prev.filter(todo => todo.id !== todoId));
+    } catch (err) {
+      setErrorMessage(ErrorType.DeleteTodo);
+      throw err;
+    } finally {
+      setLoadingTodoIds(prev => prev.filter(id => id !== todoId));
+      if (inputRef.current) {
+        inputRef.current.focus();
+      }
+    }
+  };
+
+  const handleClearCompleted = async () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    setLoadingTodoIds(prev => [
+      ...prev,
+      ...completedTodos.map(todo => todo.id),
+    ]);
+
+    try {
+      const successfulDeletes: number[] = [];
+
+      await Promise.all(
+        completedTodos.map(todo =>
+          deleteTodo(todo.id)
+            .then(() => {
+              successfulDeletes.push(todo.id);
```

Comment:
If you want to handle error cases it's better to use `Promise.allSettled()` instead
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled